docs/sdk-configuration.md

```diff
@@ -0,0 +1,676 @@
+then framework developers, then maintainers. This is because we expect those with less domain
+knowledge about tracing to require a simpler experience than those with more. There is also more
+bang-for-the-buck by making the end-user experience as streamlined as possible since we expect there
+to be much more of them than other audiences.
+
+## Goals and non-goals
+
+### Goals
+
+- Provide a single entrypoint to configuring the SDK. For end-users, less familiar with the SDK, we
+want to have everything together to provide discoverability and simpler end-user code. If there are
+several, clear use cases which benefit from different entrypoints, we could have multiple
+corresponding to each one.
+
+- Fit well with common Java idioms such as dependency injection, or common frameworks like Spring.
+
+- Reduce the chance of gotchas or configuration mishaps.
+
+- Aim for optimal performance of the configured SDK for the most common use case.
+
+### Non-goals
+
+- Provide the best possible experience for custom SDKs. Generally the burden of the experience for
+custom SDKs can fall on their authors, we optimize for our standard usage, the full SDK. Any
+reference to "the SDK" in this document refers to the full SDK with all signals.
+
+- Make sure everything is auto-configurable. This is out of the scope of the SDK, and instead is
+left to auto-configuration layers, which are also described below but not as part of the core SDK. 
+In partiular, SignalProvider SPIs which currently exist are proposed to be removed from the SDK. The 
+SDK may provide an autoconfiguration extension as an option which is not internal to the main SDK 
+components.
+
+## Configuring an instance of the SDK
+
+The SDK exposes configuration options for all the signals it supports. Users all have different
+requirements for how they use the SDK; for example they may use different exporters depending on
+their backend. Because we cannot guess the configuration the user needs, we expect that the SDK must
+be configured by the user before it can be used.
+
+Goals for configuring the SDK are
+
+- Discoverability of options
+- Ease of use by end users, e.g., less complicated code required
+- Avoid requiring duplicate configuration, which can lead to errors or confusion
+- Provide good defaults where possible
+
+In Java, the builder pattern is common for configuring instances. Let's look at what that may look
+like. The simplest configuration will be when a user wants to get a default experience, exporting
+with a specific exporter to an endpoint.
+
+The SDK builder will simply allow accept its components as builder parameters. It only allows
+setting SDK implementations and is not meant for use with partial SDKs.
+
+```java
+class OpenTelemetrySdkBuilder {
+  public OpenTelemetrySdkBuilder setTracerProvider(TracerSdkProvider tracerProvider);
+  public OpenTelemetrySdkBuilder setMeterProvider(MeterSdkProvider meterProvider);
+  public OpenTelemetrySdk buildAndSetAsGlobal();
```

Comment:
@jkwatson I had removed the auto-set as global due to controversy, but do think it's the best option in most cases. If the problem is mystery and surprise, then this naming, while obtuse, can make it clear. I would not have a normal `build()` method as that would reduce the discoverability of this mechanism - it has to be `buildWithoutSetting` or something similar to make sure a user makes a conscious decision without just assuming to use `build()`.

Reply:
the only bit that might be confusing is that the 2nd one *wouldn't* set it as global, right?

Reply:
Yeah - I think you're saying we need to document these methods well?

Reply:
lol. yes, and maybe the 2nd call to `buildAndSetAsGlobal()` would fail? Not sure what the should be, but that can be decided at a later date.